### ISSUE TITLE
Slight Delay on the figures at 9pm

### ISSUE THREAD

**Issue by benfig1127** [benfig1127/Summer-Research-2020#32]
@mikeizbicki Hey Mike, I'm working on getting the figures up for you, but it seems that some of the old data corrupted so I need to re-calculate the EN-ES M matrix for the histogram figure. It shouldn't take too long though cause I optimized the runtime of the code. I am currently finishing up the T-SNE plots and doing some organization so you dont have to look at the rats nest that is my server account. ETA: probably around 10pm for everything perfectly formatted. Sorry for the delay.

**Comment by mikeizbicki** [comment]
No problem.  Thanks for the update.

**Comment by benfig1127** [comment]
I just added the tsne plot, and it seems to pretty decent dpi. I am able to full screen and zoom in quite a bit before it becomes noticeably blurry, let me know your thoughts

**Comment by benfig1127** [comment]
@mikeizbicki (also is the '@' necessary or do you see it regardless) I know the that the left and bottom part of the ROC curve should be exactly 1.0 and 1.0, but since almost all of the curves are so near the corner it seems like a waste of space to not "zoom in" on the portions, or at least include some buffer on the margins to make the individual curve movements clearer, thoughts?

**Comment by mikeizbicki** [comment]
@benfig1127 I just realized I was having a massive brain fart about the ROC curve.  The y-axis labeled precision should be the True Positives, and the x-axis labeled recall should be the False Positive rate.  See this example: https://en.wikipedia.org/wiki/Receiver_operating_characteristic#/media/File:Roccurves.png

What I was having you plot is something related and useful, but not a ROC curve.  Could you make a plot with the adjusted axes?  I think this should be a pretty quick process of just changing the code.  I'm not sure which one will be included in the final paper, it'll depend on what the actual ROC curve looks like.

Really sorry about the confusion... I need to get more sleep :)

As for the `@`, since I'm "watching" your repo, I get an email about all the messages anyways.  But the `@` would guarantee me an email even if I weren't watching.

**Comment by benfig1127** [comment]
Yeah of course, I'll make the changes tonight wow so all the problems we were having was cause I was doing the wrong plot 😂.  Also I could only get 10 min diag entries for the histogram for now. The code already is gonna take another hour or so (4 hours in total) so for the final plot I will need to run it overnight most likely. But that plot should fall in within the hour as well.

**Comment by benfig1127** [comment]
Just to clarify, should I continue to count true negatives as true postitves for the 'new' roc curve?

**Comment by mikeizbicki** [comment]
No, that's only for the other type of plot.

**Comment by benfig1127** [comment]
So I generated a few example plots, but the results are far different, especially with the higher missing content rates. On one hand, It makes sense to me, if we remove 99% of the data, then we cant ever really get a true positive because there are so few locations where we can even get a positive let alone a true positive. But the odd symmetry of all of the curves worries me. 
![Figure_1](https://user-images.githubusercontent.com/61296503/92689928-d522ef80-f2f4-11ea-9bd5-6705c3f6754c.png)

**Comment by mikeizbicki** [comment]
I think the number you're dividing by to get the rate is the total number of samples (which is the same for all curves).  The number you should be dividing by is the total number of positive samples (which will decrease significantly for each curve, causing the curves to get much higher).

**Comment by mikeizbicki** [comment]
similarly, to get the false positive rate, you divide the false positives by the total number of negatives.  The formulas to use are for TPR and FPR on the right hand box of the wiki article: https://en.wikipedia.org/wiki/Receiver_operating_characteristic

**Comment by benfig1127** [comment]
Ahh I see, so instead of 
```
true_positive_rate=mean(true_postive_list)
where true_positive_list=[1,0,0,1,0,...] for all samples
```
it should be 
```
true_positive_rate=#true_positives/#positives 
```
and 
```
false_positive_rate=#false_positives/#negatives
```

**Comment by mikeizbicki** [comment]
yep

**Comment by benfig1127** [comment]
Okay, so I think I fixed it, it seems too good to be true though. 
![ex](https://user-images.githubusercontent.com/61296503/92694260-a52b1a80-f2fb-11ea-8b54-5a09cd4da864.png)

**Comment by benfig1127** [comment]
Here is the code snippet I am using, but it seems that this is exactly what the wiki article says:
```
for index, dist in enumerate(missing_data_list):
				
				if dist<tau_current_val:
					model_output_list.append(1)
					
					if is_translation_list[index]==1:
						true_positive_list.append(1)
						
					
					if is_translation_list[index]==0:
						false_positive_list.append(1)
						
				else:
					model_output_list.append(0)
		
					if is_translation_list[index]==1:
						false_negative_list.append(1)
						
					if is_translation_list[index]==0:
						true_negative_list.append(1)
		
		
false_positive_rate=false_positive_list.count(1)/(false_positive_list.count(1)+true_negative_list.count(1))
true_positive_rate=true_positive_list.count(1)/(true_positive_list.count(1)+false_negative_list.count(1))
```

**Comment by mikeizbicki** [comment]
That looks right!  It's interesting that the missing data rate doesn't affect the curve, but that's something I can believe and I think explain.

It just needs the axes/scale fixed for presentation.

**Comment by benfig1127** [comment]
Fantastic, I am very happy at the moment, here is a GIF that should capture my mood perfectly, hope you enjoy it lol <div class="tenor-gif-embed" data-postid="17932554" data-share-method="host" data-width="100%" data-aspect-ratio="1.0"><a href="https://tenor.com/view/cat-cat-jam-nod-pet-kitty-gif-17932554">Cat Cat Jam GIF</a> from <a href="https://tenor.com/search/cat-gifs">Cat GIFs</a></div><script type="text/javascript" async src="https://tenor.com/embed.js"></script>

**Comment by benfig1127** [comment]
All the figures have now been added to the tex file, I am still tinkering with the Tsne plots and would like some feedback on those in particular. See the fixme for some of my notes, can we schedule a meeting sometime tomorrow, I am free all day except before 11am, 4-430pm, or 8-9pm. 

Also I am adding in the MC table accuracies as well, but because a large portion of the old data got corrupted I have to now recalculate them, eta: 1500 cpu hours, or running 60 cpus at a time about 24 hours from tonight.

**Comment by mikeizbicki** [comment]
I can do 2:30pm today.

**Comment by benfig1127** [comment]
sounds good!
